snapcraft/tests/test_commands_sign_build.py

```diff
@@ -0,0 +1,99 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import os
+import logging
+from unittest import mock
+
+import fixtures
+
+from snapcraft.main import main
+from snapcraft import (
+    storeapi,
+    tests,
+)
+
+from snapcraft.tests.test_commands_register_key import (
+    get_sample_key,
+    mock_snap_output
+)
+
+
+class SignBuildTestCase(tests.TestCase):
+
+    def setUp(self):
+        super().setUp()
+        self.fake_logger = fixtures.FakeLogger(level=logging.DEBUG)
+        self.useFixture(self.fake_logger)
+        self.fake_terminal = tests.fixture_setup.FakeTerminal()
+        self.useFixture(self.fake_terminal)
+
+    @mock.patch.object(storeapi.SCAClient, 'register_key')
+    @mock.patch.object(storeapi.SCAClient, 'get_account_information')
+    @mock.patch.object(storeapi.StoreClient, 'login')
+    @mock.patch('subprocess.check_output')
+    @mock.patch('builtins.input')
+    @mock.patch('getpass.getpass')
+    @mock.patch('snapcraft.internal.repo.is_package_installed')
+    def test_sign_build_saved(self, mock_installed, mock_check_output,
+                              mock_input, mock_getpass, mock_login,
+                              mock_get_account_information,
+                              mock_register_key):
+        account_info = {'account_id': 'abcd',
+                        'account_keys': [
+                            {
+                                'name': 'default',
+                                'public-key-sha3-384': (
+                                    get_sample_key('default')['sha3-384']),
+                            },
+                        ],
+                        'snaps': {'16': {
+                            'basic': {
+                                'snap-id': 'snap-id',
+                                }
+                            }
+                        }}
+
+        mock_installed.return_value = True
+        mock_input.side_effect = ['sample.person@canonical.com', '123456']
+        mock_getpass.return_value = 'secret'
+        mock_check_output.side_effect = mock_snap_output
```

Comment:
Do you actually need this?  `mock_snap_output` only handles `snap keys --json` and `snap export-key ...`, neither of which seems relevant to `sign-build`.  I guess that's just because you're registering a key below, but that also seems somewhat irrelevant in a unit test of `sign-build`'s behaviour.

It will probably simplify your life if you mock `snap sign-build` independently (i.e. by writing your own version of `mock_snap_output` rather than modifying the one in `snapcraft.tests.test_commands_register_key`), and make your unit test independent of key registration.